server/src/main/java/io/unitycatalog/server/persist/dao/PropertyDAO.java

```diff
@@ -20,22 +22,43 @@
 @ToString
 @Builder
 public class PropertyDAO {
-
     @Id
     @UuidGenerator
     @Column(name = "id", updatable = false, nullable = false)
     private UUID id;
 
-    @Column(name = "entity_id", nullable = false)
-    private UUID entityId;
-
-    @Column(name = "entity_type", nullable = false)
-    private String entityType;
-
     @Column(name = "property_key", nullable = false)
     private String key;
 
     @Column(name = "property_value", nullable = false)
     private String value;
 
+    @ManyToOne
+    @JoinColumn(name = "catalog_id", referencedColumnName = "id")
+    private CatalogInfoDAO catalog;
```

Comment:
I'm worried that the properties schema would need to change as we add more entities with properties. Why not stick with `entity_id` and `entity_type` instead?

Reply:
If we go that way, we would have to manually create these relationships i.e. manually cascade inserts/updates/deletes instead of the entities doing it out of the box for us. This is happening currently as you can see [here](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L151) and [here](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L291) for example but I feel it is very error prone.

Reply:
Yep, a better way is to encode entity_type as part of entity ID and use it throughout the persistent layer for FK/PK reference. The earlier we do this, the less havoc we'll have in the future.

Reply:
For now, I've changed it back to a single entity_id, we will take up combining id and type in a follow up PR